deepchecks/checks/distribution/preprocessing.py

```diff
@@ -20,70 +20,73 @@
 from deepchecks.checks.distribution.rare_category_encoder import RareCategoryEncoder
 
 
-__all__ = ['preprocess_dataset_to_scaled_numerics']
+__all__ = ['ScaledNumerics']
 
 
-def preprocess_dataset_to_scaled_numerics(
-    baseline_features: pd.DataFrame,
-    test_features: pd.DataFrame,
-    categorical_columns: List[Hashable],
-    max_num_categories: int
-) -> Tuple[pd.DataFrame, pd.DataFrame]:
+class ScaledNumerics:
     """Preprocess given features to scaled numerics.
 
     Args:
         baseline_features (DataFrame):
-            Will be used for fit and also transformed. Expect to get only features
-        test_features (DataFrame):
-            Will be transformed according to baseline_data. Expect to get only features
+            Will be used for fit. Expect to get only features
         categorical_columns (List[Hashable]):
             Indicates names of categorical columns in both DataFrames.
         max_num_categories (int):
             Indicates the maximum number of unique categories in a single categorical column
             (rare categories will be changed to a form of "other")
-
-    Returns:
-        (DataFrame, DataFrame): returns both datasets transformed.
     """
-    x_baseline = baseline_features.copy()
-    x_test = test_features.copy()
-    non_categorical_columns = list(set(test_features.columns) - set(categorical_columns))
 
-    # Impute all-nan cols to all-zero:
-    def impute_whole_series_to_zero(s: pd.Series):
+    def __init__(self, baseline_features: pd.DataFrame, categorical_columns: List[Hashable], max_num_categories: int):
```

Comment:
maybe better to have this follow the way scikit learn does that. Look for e.g. at https://github.com/scikit-learn/scikit-learn/blob/7e1e6d09bcc2eaeba98f7e737aac2ac782f0e5f1/sklearn/preprocessing/_data.py#L632 and what it inherits from.
Long story short - categorical_columns: List[Hashable], max_num_categories defined in init, fitting done in fit(), transforming in transform(), fit_transform() does both one after the other.